plugins/janus_sip.c

```diff
@@ -4098,20 +4098,30 @@ static void *janus_sip_handler(void *data) {
 				/* To put the call on-hold, we need to set the direction to recvonly:
 				 * resuming it means resuming the direction we had before */
 				session->media.on_hold = hold;
+				json_t *media_direction = json_object_get(root, "media_direction");
+				const char *media_direction_text = json_string_value(media_direction);
```

Comment:
There should be a new validation parameter here, as other methods to via `JANUS_VALIDATE_JSON_OBJECT`. As it is now, this parameter is never validated anywhere, and is not guaranteed to either exist or be a string at all.